lib/shared/drive-constraints.js

```diff
@@ -208,3 +208,85 @@ exports.isDriveValid = (drive, image) => {
 exports.isDriveSizeRecommended = (drive, image) => {
   return _.get(drive, [ 'size' ], UNKNOWN_SIZE) >= _.get(image, [ 'recommendedDriveSize' ], UNKNOWN_SIZE);
 };
+
+const feedbackMessages = {
+  NOTREC: 'Not Recommended',
+  TOOSMALL: 'Too Small For Image',
+  LOCKED: 'Locked',
+  SYSTEM: 'System Drive',
+  SOURCE: 'Drive Contains Image'
+};
+const FEEDBACK_ERROR = 'error';
+const FEEDBACK_WARNING = 'warning';
+
+/**
+ * @summary Get drive feedback in an object, if any
+ * @function
+ * @public
+ *
+ * @description
+ * If the drive-image combination results in a warning or error, this function
+ * returns an object containing the feedback type ('error', 'warning') and the
+ * feedback message.
```

Comment:
Hmm, it does feel more consistent but `null` might be simpler to deal with? I'm not sure here.